test/commands/deploy.test.ts

```diff
@@ -381,6 +530,89 @@ describe('local deploy environment', function () {
+    .it('Create a local deploy with a basic component and a wildcard values file', ctx => {
+      const runCompose = Deploy.prototype.runCompose as sinon.SinonStub;
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.a_required_key).to.equal('some_value');
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.another_required_key).to.equal('required_value');
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.one_more_required_param).to.equal('one_more_value');
+    })
+
+  test
+    .timeout(15000)
+    .stub(ComponentConfigBuilder, 'buildFromPath', () => {
+      return ComponentConfigBuilder.buildFromJSON(local_component_config_with_parameters);
+    })
+    .stub(Deploy.prototype, 'readValuesFile', () => {
+      return stacked_parameter_values;
+    })
+    .stub(Deploy.prototype, 'runCompose', sinon.stub().returns(undefined))
+    .stdout({ print })
+    .stderr({ print })
+    .command(['deploy', '-l', './examples/hello-world/architect.yml', '-i', 'test:hello', '-v', './examples/hello-world/values.yml'])
+    .it('Create a local deploy with a basic component and a stacked values file', ctx => {
+      const runCompose = Deploy.prototype.runCompose as sinon.SinonStub;
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.a_required_key).to.equal('some_value');
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.another_required_key).to.equal('required_value');
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.one_more_required_param).to.equal('one_more_value');
+    })
+
+  test
+    .timeout(15000)
+    .stub(ComponentConfigBuilder, 'buildFromPath', () => {
+      return ComponentConfigBuilder.buildFromJSON(local_component_config_with_dependency);
+    })
+    .stub(Deploy.prototype, 'readValuesFile', () => {
+      return component_and_dependency_parameter_values;
+    })
+    .nock(MOCK_API_HOST, api => api
+      .get(`/accounts/examples/components/react-app/versions/latest`)
+      .reply(200, local_component_config_dependency))
+    .stub(Deploy.prototype, 'runCompose', sinon.stub().returns(undefined))
+    .stdout({ print })
+    .stderr({ print })
+    .command(['deploy', '-l', './examples/hello-world/architect.yml', '-i', 'test:hello', '-v', './examples/hello-world/values.yml'])
+    .it('Create a local deploy with a basic component, a dependency, and a values file', ctx => {
+      const runCompose = Deploy.prototype.runCompose as sinon.SinonStub;
+      expect(runCompose.firstCall.args[0].services['examples--hello-world--api--latest--d00ztoyu'].environment.a_required_key).to.equal('some_value');
```

Comment:
Try not to hardcode the generated refs... If they were to change it'll be a pain to update all of the tests.